slam_toolbox/lib/karto_sdk/src/Mapper.cpp

```diff
@@ -1659,6 +1664,40 @@ namespace karto
     return pEdge;
   }
 
+  kt_bool MapperGraph::AddEdge(Edge<LocalizedRangeScan> * pEdge)
+  {
+    using IteratorT = std::map<int, Vertex<LocalizedRangeScan> *>::iterator;
+
+    LocalizedRangeScan * pSourceScan = pEdge->GetSource()->GetObject();
+    LocalizedRangeScan * pTargetScan = pEdge->GetTarget()->GetObject();
+    IteratorT v1 = m_Vertices[pSourceScan->GetSensorName()]
+                   .find(pSourceScan->GetStateId());
+    IteratorT v2 = m_Vertices[pTargetScan->GetSensorName()]
+                   .find(pTargetScan->GetStateId());
+
+    if (v1 == m_Vertices[pSourceScan->GetSensorName()].end() ||
+        v2 == m_Vertices[pSourceScan->GetSensorName()].end())
+    {
+      std::cout << "AddEdge: At least one vertex is invalid." << std::endl;
+      return false;
+    }
+
+    // see if edge already exists
+    for (Edge<LocalizedRangeScan> * pOtherEdge : GetEdges()) {
```

Comment:
Iterating through the entire graph eek, is this necessary?

Reply:
Or can this be error checked before now not requiring the check here?

Reply:
Hmm, yeah, this will degrade performance, and it's just a sanity check. Logic in `Mapper::MarginalizeNodeFromGraph()` below operates on the assumption that `edge already exists` will never be true when using this `MapperGraph::AddEdge` overload. 

Maybe we can put this within an `#ifdef _DEBUG` block. WDYT?

Reply:
If it should never be true, I would just remove it.

Reply:
I have not, but I did compile for debug and there's an assertion below. It didn't ever abort (there would be a major logic error if it did).

Reply:
OK - please remove then. That would be nontrivial to deal with

Reply:
Alright, dropped in https://github.com/SteveMacenski/slam_toolbox/pull/468/commits/295cce40d5e0edb05eb404cab089fe3c6dedf1f4.